File: Question_7/Java/Question7.java
import java.io.*;
import java.util.*;
import java.util.stream.Stream;

class Solution {
    static String findSuspiciousUserId(int numOfQuestions, int[][] questionAndAnswerArrOfArr) {
        Set suspicious = new TreeSet();
        for (int[] ints : questionAndAnswerArrOfArr) {
            for (int[] value : questionAndAnswerArrOfArr) {
                if (value[0] != ints[0] && find(value, ints[0]) && find(ints, value[0])) {
                    suspicious.add(value[0]);
                }
            }
        }
        int num = 0;
        while (num != suspicious.size()) {
            num = suspicious.size();
            for (int[] ints : questionAndAnswerArrOfArr) {
                if (!suspicious.contains(ints[0])) {
                    int i = 0;
                    for (int j = 1; j < ints.length; j++) {
                        if (suspicious.contains(ints[j])) {
                            i++;
                        }
                        if (i >= 2) {
                            suspicious.add(ints[0]);
                            break;
                        }
                    }

                }
            }
        }
        StringBuilder stringBuilder = new StringBuilder();
        for (Object i : suspicious) {
            stringBuilder.append(i);
            stringBuilder.append(",");
        }
        stringBuilder.deleteCharAt(stringBuilder.length() - 1 );
        return stringBuilder.toString();
    }

    private static boolean find(int[] array, int value) {
        for (int a : array) {
            if (a == value) {
                return true;
            }
        }
        return false;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        // Sample input:
        // 3
        // 1 2,2 1,3 1 2
        String[] firstLine = scanner.nextLine().split(" ");
        String secondLine = scanner.nextLine();
        scanner.close();

        int numOfQuestions = Integer.parseInt(firstLine[0]);
        String[] questionAndAnswers = secondLine.split(",");
        int[][] questionAndAnswerArrOfArr = parseQuestionAndAnswer(questionAndAnswers);

        String result = findSuspiciousUserId(numOfQuestions, questionAndAnswerArrOfArr);
        // Please do not remove the below line.
        System.out.println(result);
        // Do not print anything after this line
    }

    private static int[][] parseQuestionAndAnswer(String[] questionAndAnswers) {
        return Arrays.stream(questionAndAnswers)
                .map(questionAndAnswer -> {
                    String[] questionAndAnswerArr = questionAndAnswer.split(" ");
                    return Stream.of(questionAndAnswerArr).mapToInt(Integer::parseInt).toArray();
                })
                .toArray(int[][]::new);
    }
}

File: Question_7/Python/Question7.py
def findSuspiciousUserId(questionAndAnswerListOfList):
    string = [""] * (max(max(x) for x in questionAndAnswerListOfList) + 1)
    for i in range(len(questionAndAnswerListOfList)):
        for value in questionAndAnswerListOfList[i + 1:]:
            if questionAndAnswerListOfList[i][0] in value and value[0] in questionAndAnswerListOfList[i]:
                string[value[0]] = str(value[0]) + ","
                string[questionAndAnswerListOfList[i][0]] = str(questionAndAnswerListOfList[i][0]) + ","
    actual = 0
    previous = -1
    while previous < actual:
        previous = actual
        for ints in questionAndAnswerListOfList:
            if not string[ints[0]]:
                i = 0
                for j in ints[1:]:
                    if string[j]:
                        i += 1
                        if i == 2:
                            string[ints[0]] = str(ints[0]) + ","
                            actual += 1
                            break
    return ("".join(string))[:-1]


def main():
    firstLine = input().split(" ")
    secondLine = input()

    # Sample input:
    # 3
    # 1 2,2 1,3 1 2

    numOfQuestions = int(firstLine[0])
    questionAndAnswers = secondLine.split(",")
    questionAndAnswerListOfList = parseQuestionAndAnswer(questionAndAnswers)

    # Participants may update the following function parameters
    answer = findSuspiciousUserId(questionAndAnswerListOfList)

    # Please do not remove the below line.
    print(answer)
    # Do not print anything after this line


def parseQuestionAndAnswer(questionAndAnswers):
    questionAndAnswerListOfList = []
    for index in range(0, len(questionAndAnswers)):
        questionAndAnswerList = questionAndAnswers[index].split(" ")
        questionAndAnswerListOfList.append([int(x) for x in questionAndAnswerList])
    return questionAndAnswerListOfList


if __name__ == '__main__':
    main()
